internal/monitors/conviva/conviva.go

```diff
@@ -169,33 +168,35 @@ func (m *Monitor) fetchMetrics(contextTimeout time.Duration, semaphore chan stru
 				logger.Errorf("GET metric %s failed. %+v", metricConf.MetricParameter, err)
 				return
 			}
+			timestamp := time.Now()
 			dps := make([]*datapoint.Datapoint, 0)
 			for metricParameter, series := range res {
 				metricConf.logFilterStatuses(series.Meta.FiltersWarmup, series.Meta.FiltersNotExist, series.Meta.FiltersIncompleteData)
 				metricName := "conviva." + metricParameter
 				for filterID, metricValues := range series.FilterIDValuesMap {
+				L:
 					for i, metricValue := range metricValues {
 						select {
 						case <-ctx.Done():
```

Comment:
This is really unnecessary since the stuff in the default case is non-blocking.  Just let it make the datapoints and don't worry about trying to catch shutdown with such fine precision.  The Done channel is mainly for stopping goroutines.